server/routes/team.router.js

```diff
@@ -0,0 +1,105 @@
+const express = require('express');
+const pool = require('../modules/pool');
+const router = express.Router();
+const {
+  rejectUnauthenticated,
+} = require("../modules/authentication-middleware");
+
+router.get('/all', (req, res) => {
+  // console.log('in teams GET router');
+  let queryText = `
+    SELECT "users".name as userName, 
+      "usersTeams"."userId", "usersTeams"."teamId", 
+      "teams".name as "teamName", "teams".id as "teamId", "teams"."captainId", "teams"."accessCode"
+    FROM "users"
+    JOIN "usersTeams" ON "users".id = "usersTeams"."userId"
+    JOIN "teams" ON "teams".id = "usersTeams"."teamId";`
+  pool
+    .query(queryText)
+    .then((result) => {
+      res.send(result.rows);
+    })
+    .catch(err => {
+      console.log('Error in Team GET', err);
+      res.sendStatus(500)
+    })
+});
+
+router.get('/access', (req, res) => {
+  let queryText=`
+    SELECT "teams".id as "ID", "teams"."accessCode" as "accessCode" 
+    FROM "teams";
+  `;
+  pool
+    .query(queryText)
+    .then((result) => {
+      res.send(result.rows);
+    })
+    .catch(err => {
+      console.log('Error getting access codes', err);
+      res.sendStatus(500)
+    })
+})
+
+router.get('/leagueTeam/:id', (req, res) => {
+  console.log('test1', req.body);
+  let queryText = `SELECT * FROM teams WHERE "teams"."leagueId" = $1`;
+  pool.query(queryText, [req.params.id])
+  .then((result) => {
+    res.send(result.rows);
+  })
+  .catch(err => {
+    console.log('error getting teams', err);
+    res.sendStatus(500)
+  })
+})
+
+router.post('/', async (req, res) => {
+  let accessCode = Math.random().toString(36).replace(/[^a-z]+/g, '').substr(0, 6).toUpperCase();
```

Comment:
cool!

If you wanted to have some fun with this, you could use a library like
https://www.npmjs.com/package/unique-names-generator

Generates a unique phrase, like `bigBlueDonkey`.